graph/db/kv_store.go

```diff
@@ -3109,6 +3109,43 @@ func (c *KVStore) ForEachNodeChannel(nodePub route.Vertex,
 	return nodeTraversal(nil, nodePub[:], c.db, cb)
 }
 
+// ForEachSourceNodeChannel iterates through all channels of the source node,
+// executing the passed callback on each. The call-back is provided with the
+// channel's outpoint, whether we have a policy for the channel and the channel
+// peer's node information.
+func (c *KVStore) ForEachSourceNodeChannel(cb func(chanPoint wire.OutPoint,
+	havePolicy bool, otherNode *models.LightningNode) error) error {
+
+	return kvdb.View(c.db, func(tx kvdb.RTx) error {
+		nodes := tx.ReadBucket(nodeBucket)
+		if nodes == nil {
+			return ErrGraphNotFound
+		}
+
+		node, err := c.sourceNode(nodes)
+		if err != nil {
+			return err
+		}
+
+		return nodeTraversal(tx, node.PubKeyBytes[:], c.db,
```

Comment:
micro-nit: insert line break before `tx` argument, since this is technically a multi-line function call that doesn't introduce a new scope/block within the first line.